### ISSUE TITLE
Discovery: BTSSS react widget request (travel pay)

### ISSUE THREAD

**Issue by wesrowe** [department-of-veterans-affairs/va.gov-cms#9206]
## Description

First issue for epic, to cover product/UX discovery for this request.

- [Benefits detail page](https://www.va.gov/health-care/get-reimbursed-for-travel-pay/) for Travel Reimbursement
- Here's the [preview of the travel pay reimbursement page with planned changes](http://preview-prod.vfs.va.gov/preview?nodeId=2431) (SOCKS required)(from Danielle's comment below)

**Ideal UX for new widget:** Get Vet to sign in, use the VA.gov authentication, and then deep-link them to the page within that platform for [submitting a claim?]

### Current State

<details><summary>Images: </summary>

- Click CTA on the Benefit Detail page: <img width="664" alt="image" src="https://user-images.githubusercontent.com/4054752/170590507-7d455c1a-bfe3-47f0-9493-a5ac9fd576d6.png">
- Sign into Veteran Travel Claim Entry: <img width="576" alt="image" src="https://user-images.githubusercontent.com/4054752/170590579-516529db-57a7-466f-ad17-6cbc0de37272.png">
- Possible further wayfinding step after login, we don't have access to verify

</details>

### Discovery:
* Get a login - check TUD first, someone eligible to get reimbursements - maybe just a health care beneficiary can sign into platform
* Node 2431 is the draft content that DT is working on; there is also a R&S article
* There’s a national-level page about travel; research findings showed VA Travel Pay is a big Top Task missing from VAMC product. So once this page and login exp is done, we’ll let Facilities figure out how to integrate this widget into their product.
* Widget will exist in a library of some sort that Drupal can access.

Technical discovery / dependencies (blocked by prioritization):
- [x] Ensure we can deep-link to the right page - **No deep-link is possible. BTSSS has required hoops user has to jump through after logging in: 1) Accept Terms, 2) review profile**
- [x] Ensure the token/cookie from VA.gov login will be accepted at the deep-link destination. - **VAgov platform/infrastructure may be working on an OAuth implementation that could help users avoid logging in twice. BTSSS has been in touch with Cody Reinold. ETA unknown.**
- [x] [Wes] to see if TUD user can get me into the current experience - **nope**

Danielle to check with 18F to see if deeplinking is viable.

## Acceptance Criteria
- [x] met with BTSSS team, received demo
- [ ] seek Dave's feedback on whether to continue pursuit

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Platform CMS Team`
- [ ] `Sitewide program`
- [ ] `⭐️ Sitewide CMS`
- [x] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by DanielleThierryUSDSVA** [comment]
Hi @wesrowe. Thanks for meeting today! Adding a couple things here:

- Here's the [preview of the travel pay reimbursement page with planned changes](http://preview-prod.vfs.va.gov/preview?nodeId=2431).

- I tried signing signing in to Access VA with my PIV. It took me here: https://dvagov-btsss.dynamics365portals.us/ (a publicly accessible page that sends you back to Access VA if you're not authenticated). Since I'm not set up in travel pay, it gave me this message:

![image](https://user-images.githubusercontent.com/62957278/170144016-173e0d6d-3578-4149-8495-62919b288470.png)

**Comment by wesrowe** [comment]
@DanielleThierryUSDSVA, can you point out where the new widget would appear on the draft/preview you shared? E.g., does it replace the CTA link " > Go to the AccessVA travel claim portal to file a claim"?

**Comment by DanielleThierryUSDSVA** [comment]
Hi @wesrowe. Yes, it would replace that CTA link. And we'd adjust the content above to remove mention of the AccessVA travel claim portal since they wouldn't have to go through that anymore.

**Comment by wesrowe** [comment]
Update 6/1: Discovery will require meeting with folks who own the travel pay portal. Danielle to introduce Wes/Aslan to a BTSSS SME (expert).

**Comment by jcklpe** [comment]
@wesrowe  this ticket is on hold until we have that BTSSS SME meeting correct?

**Comment by jcklpe** [comment]
@wesrowe : Wanted to check in on this. This has been set to block. Do move it into a specific column on the ZenHub board?

**Comment by jilladams** [comment]
We aren't currently using a specific column for Blocked issues, just noting them, and keeping them out of "In Progress" while blocked (this is sitting in New issues column).  The meeting today unblocked Discovery here, so have removed that label.

**Comment by jcklpe** [comment]
@jilladams : Okay cool. My board is showing this as "In Progress" but it still has the tag "needs refinement" and I can see some parts of the OP ticket comment that lacks details so I assume this needs to be pushed back to "New/Unrefined". I will do that, let me know if that is not the correct thing to do.

**Comment by wesrowe** [comment]
Updated Dave in Slack based on today's demo:
There doesn't appear to be any low-hanging fruit here for improving the Veteran experience. We are already linking Veterans directly to BTSSS's login page. Oauth integration could potentially save the Veteran from logging in a second time (if they logged into VAgov first). However, most of the pain of this experience occurs after they've logged into BTSSS, whereupon business rules require them to 1) accept terms and conditions, and then 2) review their profile info, before 3) finally landing on the dashboard where they do the claim process(es). We can discuss further details if you're interested.

**Comment by wesrowe** [comment]
@jcklpe, thanks for your care but please don't move issues back to new/unrefined. Jill and I are responsible for quality of issues in sprint, so you can bring any concerns to our attention.

**Comment by jcklpe** [comment]
Notes from BTSSS meeting:

> 6/22 Demo of BTS3 portal, current experience
> Agenda:
>   1. What is the Veteran experience today when reaching BTSSS from the related VA.gov benefit detail page<https://www.va.gov/health-care/get-reimbursed-for-travel-pay/>?
>   2. Does BTSSS have the deep-link capability to take a Veteran directly to making a claim (after logging into BTSSS)?
> 
> Flow summary
> Va.gov > [health care benefit hub](https://staging.va.gov/health-care/)
> 
> Find travel pay reimbursement link:
> ![image](https://user-images.githubusercontent.com/9451726/175343194-620d11e3-94d1-45a9-b616-434e938b7878.png)
> 
> Find link to AccessVA website:
>  ![image](https://user-images.githubusercontent.com/9451726/175343316-2254aa5f-def0-4fe3-9846-2f89921188e3.png)
> 
> 
> Click your role (I chose Veteran), and then click correct tile from options (at bottom left):
> ![image](https://user-images.githubusercontent.com/9451726/175343441-6e026c65-6591-4bae-9a89-ebd7cfcf95ae.png)

> 
> Select login method:
>  ![image](https://user-images.githubusercontent.com/9451726/175343534-936fb80a-d6b0-4b19-a25a-1552e197b80d.png)

> ![image](https://user-images.githubusercontent.com/9451726/175343611-92e6f5ca-ea84-49a1-9110-f940ed4fb84a.png)

> 
> 
> After some SAML screen transitions…
>   ![image](https://user-images.githubusercontent.com/9451726/175343702-c2d706b2-f9b1-48c3-855e-7d76f91b447d.png)

> Profile Review screen
> (note: “Request profile update” is the submit button on this page) 
>  ![image](https://user-images.githubusercontent.com/9451726/175343775-eaf933ac-1655-47ff-a871-dd61d549f4e3.png)

> Click: proceed to dashboard. Dashboard columns are sortable.
>  ![image](https://user-images.githubusercontent.com/9451726/175343851-62ff9a7a-9941-4819-a438-f10ef64cc4e2.png)

> Dashboard is where the Vet can finally take action.
> 
> Aaron jumps in to show how OAuth changes things – 
> With OAuth, the best we can do is link a logged-in Vet directly to the T&C page at top.
> Timing: late summer / early fall - Sept to production
> 
> Attendees:
> Deepika Lambda
> Michelle Ortiz
> Aaron Kauffman
> Danielle Thierry
> (As well as PW team: Wes, me, Ryan, Aslan, Florence)
> Deep Linking options
> Per Aaron, the current auth process (SAML) only allows us to link to the MSFT login screen.
> Only technical limitation is only having this issue w/ authorization
> Other complexity driven by business rules 
> After sign in, users are required to ack the Terms & Conditions, and review Profile data before proceeding to a user dashboard where they can file claims or create appointments. 
> Those 2 screens (T&C and Profile) are in place for business reasons that will require approval to bypass. 
> BTSSS is in process of implementing OAuth.
> Timing: late summer / early fall (~Sept) to ship in production
> After OAuth ships, the best we can do is link a logged-in Vet directly to the T&C page at top.
> 
> Next steps are TBD. Wesley Rowe Do we:
> Implement short term link to login
> Wait til OAuth
> Review Discovery options with Dave and he decides

And I'm going to close this ticket, as I understand this work is not going to be taken further.

**Comment by jilladams** [comment]
@jcklpe updated formatting for legibility and added the pointer to where Wes has proposed next steps to Dave in Slack. That will be the determining factor in whether this work moves forward in some way. (But you're right that discovery is done.)
